lib/sensors/temperature.cpp

```diff
@@ -2,45 +2,63 @@
 
 namespace hyped::sensors {
 
-Temperature::Temperature(hyped::io::I2c &i2c, hyped::core::ILogger &log) : log_(log), i2c_(i2c)
+Temperature::Temperature(const std::uint8_t mux_address,
+                         const std::uint8_t channel,
+                         hyped::io::I2c &i2c,
+                         hyped::core::ILogger &log)
+    : log_(log),
+      i2c_(i2c),
+      channel_(channel),
+      mux_address_(mux_address)
+{
+}
+
+Temperature::~Temperature()
 {
 }
 
 std::optional<std::uint16_t> Temperature::read()
 {
-  const auto Status_check = i2c_.readByte(kMux, kStatus);
-
-  if (Status_check) {
-    if (Status_check.value() == 0) {
-      const std::optional<std::uint8_t> tempurature_high = i2c_.readByte(kMux, kTempuratureHigh);
-      const std::optional<std::uint8_t> tempurature_low  = i2c_.readByte(kMux, kTempuratureLow);
-      // getting the high and low values from the register
-
-      if (tempurature_high.has_value() && tempurature_low.has_value()) {
-        std::uint16_t tempurature = (tempurature_high.value() << 8) | tempurature_low.value();
-        tempurature               = tempurature * 0.01;
-
-        return tempurature;
-      } else {
-        log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
-        return std::nullopt;
-      }
-      // Reading from the Status register to see if the tempurature is ready to be read
-      // Need to have it do the correct effect and output a log of what occurred
-
-    } else {
-      log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
-      return std::nullopt;
-    }
-  } else {
-    log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+  const auto status_check_result = i2c_.readByte(mux_address_, kStatus);
+
+  const auto temperature_high = i2c_.readByte(mux_address_, kData_T_H);
+  const auto temperature_low  = i2c_.readByte(mux_address_, kData_T_L);
+
+  // getting the temperature_high_byte and temperature_low_byte values from the register
+
+  if (!status_check_result) {
```

Comment:
this should move right below the read

you should do the read and then immediately check that it was successful, always.